lib/collection/src/shards/local_shard/clock_map.rs

```diff
@@ -191,6 +202,7 @@ impl RecoveryPoint {
     pub fn iter_as_clock_tags(&self) -> impl Iterator<Item = ClockTag> + '_ {
         self.clocks
             .iter()
+            // TODO: keep clock token here? currently we generate a new unique one.
```

Comment:
Important: when updating the cutoff point we currently lose the tokens. In this place we simply generate a new unique one. Though not strictly necessary, we likely want to use the original tokens here.

Reply:
I have added this now: recovery points also hold a unique token.

It is not used for resolving the WAL delta. But it is used to set the cutoff point.